File: java/selection_sort.java
public class SelectionSort
{
	/*
	SelectionSort Algorithm Implementation in Java
	time complexity : O(n^2)
	*/ 
	static int[] selectionSort(int[] arr)
	{
		int size = arr.length;

		for (int i = 0 ;i< size-1; i++)
		{
			int min = i;

			for (int j = i+1; j<size; j++)
			{
				if (arr[j] < arr[min])
				{
					min = j;
				}
			}
		 
			int temp = arr[min];
			arr[min] = arr[i];
			arr[i] = temp;
		}
		
		return arr;
	}
	
	
	//Driver main method
   public static void main(String args[])
   {
		int[] arr = {5,1,3,6,8,12,4};   
		
		arr=selectionSort(arr);
		
		for (int i = 0 ;i< arr.length; i++)
		{
			System.out.print(arr[i] + " ");
		}
   }  
}

File: python/selection_sort.py
# Python program for implementation of Selection 
def selectionSort(array):
  n = len(array)
  for i in range(n):
    # Initially, assume the first element of the unsorted part as the minimum.
    minimum = i

    for j in range(i+1, n):
      if (array[j] < array[minimum]):
        # Update position of minimum element if a smaller element is found.
        minimum = j

    # Swap the minimum element with the first element of the unsorted part.     
    temp = array[i]
    array[i] = array[minimum]
    array[minimum] = temp
    
  return array
  
# Driver code to test above 
print("Enter array elements")
array = list(map(int,input().split()))
print ("Sorted array is ",selectionSort(array)) 



